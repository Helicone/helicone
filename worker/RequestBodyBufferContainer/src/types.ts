export type StoredEntry = {
  data: Buffer;
  size: number;
  expiresAt: number; // epoch millis
};

export type SignAwsInput = {
  region: string;
  forwardToHost: string;
  requestHeaders: Record<string, string>;
  method: string;
  urlString: string;
};

export type SignAwsOutput = {
  newHeaders: Record<string, string>;
  model: string;
};

