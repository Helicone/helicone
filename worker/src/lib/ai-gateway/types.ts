import { Endpoint } from "@helicone-package/cost/models/types";
import { ProviderKey } from "../db/ProviderKeysStore";
import { ModelProviderName } from "@helicone-package/cost/models/providers";

export interface Attempt {
  endpoint: Endpoint;
  providerKey: ProviderKey;
  authType: "byok" | "ptb";
  priority: number;
  source: string;
}

export interface ModelSpec {
  modelName: string;
  provider?: ModelProviderName;
  customUid?: string;
}

export interface DisallowListEntry {
  provider: string;
  model: string;
}

// For backwards compatibility
export interface EscrowInfo {
  escrowId: string;
  endpoint: Endpoint;
}

// Error type matching the old aiGateway pattern
export type AttemptError = {
  type:
    | "invalid_format"
    | "missing_provider_key"
    | "request_failed"
    | "invalid_prompt"
    | "model_not_supported";
  message: string;
  statusCode: number;
  details?: string;
};
