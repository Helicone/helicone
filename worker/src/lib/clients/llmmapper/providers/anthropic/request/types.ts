export interface AntRequestBody {
	model: string;
	messages: {
		role: 'user' | 'assistant' | 'system';
		content: string | ContentBlock[];
	}[];
	max_tokens: number;
	metadata?: {
		user_id?: string;
	};
	system?: string;
	temperature?: number;
	top_p?: number;
	top_k?: number;
	stop_sequences?: string[];
	stream?: boolean;
}

export interface ContentBlock {
	type: 'text' | 'image';
	text?: string;
	source?: {
		type: 'base64' | 'url';
		media_type: string;
		data: string;
	};
}
