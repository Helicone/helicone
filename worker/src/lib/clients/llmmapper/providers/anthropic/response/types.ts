export interface AntResponseBody {
	id: string;
	type: 'message';
	role: 'assistant';
	content: ContentBlock[];
	model: string;
	stop_reason: 'end_turn' | 'max_tokens' | 'stop_sequence' | 'tool_use' | null;
	stop_sequence: string | null;
	usage: {
		input_tokens: number;
		output_tokens: number;
	};
}

export interface ContentBlock {
	type: 'text' | 'tool_use';
	text?: string;
	id?: string;
	name?: string;
	input?: object;
}
