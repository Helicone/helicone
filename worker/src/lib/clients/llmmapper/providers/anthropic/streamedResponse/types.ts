/* eslint-disable @typescript-eslint/no-explicit-any */
// Base event type
export interface BaseEvent {
  type: string;
}

// Message start event
export interface MessageStartEvent extends BaseEvent {
  type: "message_start";
  message: {
    id: string;
    type: "message";
    role: "assistant";
    content: Array<{
      type: "text" | "tool_use";
      text?: string;
      id?: string;
      name?: string;
      input?: Record<string, any>;
    }>;
    model: string;
    stop_reason: string | null;
    stop_sequence: string | null;
    usage: {
      input_tokens: number;
      output_tokens: number;
    };
  };
}

// Content block start event
export interface ContentBlockStartEvent extends BaseEvent {
  type: "content_block_start";
  index: number;
  content_block: {
    type: "text" | "tool_use";
    text?: string;
    id?: string;
    name?: string;
    input?: Record<string, any>;
  };
}

export interface ContentBlockDeltaEvent extends BaseEvent {
  type: "content_block_delta";
  index: number;
  delta:
    | {
        type: "text_delta";
        text: string;
      }
    | {
        type: "input_json_delta";
        partial_json: string;
      };
}

// Content block stop event
export interface ContentBlockStopEvent extends BaseEvent {
  type: "content_block_stop";
  index: number;
}

// Message delta event
export interface MessageDeltaEvent extends BaseEvent {
  type: "message_delta";
  delta: {
    stop_reason: string | null;
    stop_sequence: string | null;
  };
  usage?: {
    output_tokens?: number;
    input_tokens?: number;
  };
}

// Message stop event
export interface MessageStopEvent extends BaseEvent {
  type: "message_stop";
}

// Ping event
export interface PingEvent extends BaseEvent {
  type: "ping";
}

// Error event
export interface ErrorEvent extends BaseEvent {
  type: "error";
  error: {
    type: string;
    message: string;
  };
}

// Union type for all possible events
export type AnthropicStreamEvent =
  | MessageStartEvent
  | ContentBlockStartEvent
  | ContentBlockDeltaEvent
  | ContentBlockStopEvent
  | MessageDeltaEvent
  | MessageStopEvent
  | PingEvent
  | ErrorEvent;
