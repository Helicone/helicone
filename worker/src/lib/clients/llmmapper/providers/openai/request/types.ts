export interface OpenAIRequestBody {
	model: string;
	messages: {
		role: 'system' | 'user' | 'assistant' | 'function';
		content:
			| string
			| null
			| Array<{
					type: 'text' | 'image_url';
					text?: string;
					image_url?: {
						url: string;
						detail?: 'auto' | 'low' | 'high';
					};
			  }>;
		name?: string;
		function_call?: {
			name: string;
			arguments: string;
		};
	}[];
	functions?: {
		name: string;
		description?: string;
		parameters: object; // JSON Schema object
	}[];
	function_call?: 'auto' | 'none' | { name: string };
	temperature?: number;
	top_p?: number;
	n?: number;
	stream?: boolean;
	stop?: string | string[];
	max_tokens?: number;
	presence_penalty?: number;
	frequency_penalty?: number;
	logit_bias?: { [key: string]: number };
	user?: string;
	seed?: number;
	tools?: {
		type: string;
		function: {
			name: string;
			description?: string;
			parameters: object; // JSON Schema object
		};
	}[];
	tool_choice?: 'auto' | 'none' | { type: string; function: { name: string } };
	response_format?: { type: 'text' | 'json_object' };
}
