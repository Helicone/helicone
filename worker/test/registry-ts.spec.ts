import { SELF, fetchMock } from "cloudflare:test";
import { describe, it, expect, beforeAll, afterAll } from "vitest";
import { registry } from "@helicone-package/cost/models/registry";
import { UserEndpointConfig } from "@helicone-package/cost/models/types";
import "./setup";
import { type TestCase } from "./providers/base.test-config";
import { anthropicTestConfig } from "./providers/anthropic.test-config";

function mockRequiredServices() {
  const callTrackers = {
    s3Called: false,
    loggingCalled: false,
  };

  const s3Mock = fetchMock
    .get("http://localhost:9000")
    .intercept({
      path: /.*/,
      method: "PUT",
    })
    .reply(() => {
      callTrackers.s3Called = true;
      return { statusCode: 200, data: "" };
    })
    .persist();

  const loggingMock = fetchMock
    .get("http://localhost:8585")
    .intercept({
      path: "/v1/log/request",
      method: "POST",
    })
    .reply(() => {
      callTrackers.loggingCalled = true;
      return { statusCode: 200, data: { success: true } };
    })
    .persist();
  return { s3Mock, loggingMock, callTrackers };
}

function mockProviderEndpoint(
  modelId: string,
  provider: string,
  statusCode: number,
  byokConfig: UserEndpointConfig = {}
) {
  const config = registry.getModelProviderConfig(modelId, provider).data;
  if (!config) return;

  const endpoint = registry.buildEndpoint(config, byokConfig).data;
  if (!endpoint) return;

  const url = new URL(endpoint.baseUrl);

  if (statusCode === 200) {
    // For now just use anthropic config, in future we'll have a map of providers
    const testConfig = anthropicTestConfig;
    
    fetchMock
      .get(`${url.protocol}//${url.host}`)
      .intercept({ path: url.pathname, method: "POST" })
      .reply((request) => {
        const body = JSON.parse(request.body as string);
        const modelName = body.model?.split("/")[0] || body.model;
        return {
          statusCode: 200,
          data: testConfig.generateMockResponse(modelName),
          responseOptions: { headers: { "content-type": "application/json" } },
        };
      })
      .persist();
  } else {
    fetchMock
      .get(`${url.protocol}//${url.host}`)
      .intercept({ path: url.pathname, method: "POST" })
      .reply(() => ({ statusCode, data: { error: "Provider failed" } }))
      .persist();
  }
}

function setupTestMocks(testCase: TestCase) {
  const byokConfig = testCase.byokConfig || {};

  if (testCase.testType === "multi-provider-fallback") {
    // Mock failures for specified providers (with same byokConfig)
    testCase.failProviders?.forEach((failProvider) => {
      mockProviderEndpoint(testCase.modelId, failProvider, 500, byokConfig);
    });
  }

  // Mock success for the target provider
  if (testCase.provider) {
    mockProviderEndpoint(testCase.modelId, testCase.provider, 200, byokConfig);
  }
}

describe("Registry Tests", () => {
  beforeAll(() => {
    fetchMock.activate();
    fetchMock.disableNetConnect();
    mockRequiredServices();
  });

  afterAll(() => {
    fetchMock.deactivate();
  });

  // PTB Tests commented out - PTB is not currently active (see aiGateway.ts line 241)
  // describe("PTB Tests", () => {
  //   const ptbTestCases: TestCase[] = [...anthropicTestConfig.generatePtbTestCases()];

  //   ptbTestCases.forEach((testCase) => {
  //     it(testCase.name, async () => {
  //       if (!testCase.provider) {
  //         console.warn(`No provider specified for ${testCase.name}`);
  //         return;
  //       }
  //       const endpointsResult = registry.getPtbEndpointsByProvider(
  //         testCase.modelId,
  //         testCase.provider
  //       );
  //       const endpoints = endpointsResult.data || [];

  //       endpoints.forEach((endpoint) => {
  //         const url = new URL(endpoint.baseUrl);
  //         const baseUrl = `${url.protocol}//${url.host}`;
  //         const path = url.pathname;

  //         fetchMock
  //           .get(baseUrl)
  //           .intercept({
  //             path: path,
  //             method: "POST",
  //           })
  //           .reply((request) => {
  //             const body = JSON.parse(request.body as string);
  //             const modelName = body.model?.split("/")[0] || body.model;
  //             return {
  //               statusCode: 200,
  //               data: generateMockResponse(modelName),
  //               responseOptions: {
  //                 headers: { "content-type": "application/json" },
  //               },
  //             };
  //           })
  //           .persist();
  //       });

  //       const response = await SELF.fetch(
  //         "https://ai-gateway.helicone.ai/v1/chat/completions",
  //         {
  //           method: "POST",
  //           headers: {
  //             "Content-Type": "application/json",
  //             Authorization:
  //               "Bearer sk-helicone-aaa1234-bbb1234-ccc1234-ddd1234",
  //           },
  //           body: JSON.stringify({
  //             model: `${testCase.modelId}/${testCase.provider}`,
  //             ...testCase.request,
  //           }),
  //         }
  //       );

  //       expect(response.status).toBe(200);
  //       const body = (await response.json()) as any;
  //       expect(body).toHaveProperty("model");
  //       expect(body).toHaveProperty("usage");
  //       expect(body.usage.total_tokens).toBe(
  //         body.usage.prompt_tokens + body.usage.completion_tokens
  //       );
  //     });
  //   });
  // });

  describe("BYOK Tests", () => {
    const byokTestCases: TestCase[] = [
      ...anthropicTestConfig.generateByokTestCases(),
      // Future: ...bedrockTestConfig.generateByokTestCases(),
    ];

    byokTestCases.forEach((testCase) => {
      it(testCase.name, async () => {
        setupTestMocks(testCase);

        const response = await SELF.fetch(
          "https://ai-gateway.helicone.ai/v1/chat/completions",
          {
            method: "POST",
            headers: {
              "Content-Type": "application/json",
              Authorization:
                "Bearer sk-helicone-aaa1234-bbb1234-ccc1234-ddd1234",
            },
            body: JSON.stringify({
              model: testCase.modelString,
              ...testCase.request,
            }),
          }
        );

        expect(response.status).toBe(200);
        const body = (await response.json()) as any;
        expect(body).toHaveProperty("model");
        expect(body).toHaveProperty("usage");
        expect(body.usage.total_tokens).toBe(
          body.usage.prompt_tokens + body.usage.completion_tokens
        );
      });
    });
  });

  describe("Customer-Specific Tests", () => {
    
    describe("Anthropic SDK with NO_MAPPING", () => {
      it("should route to /v1/messages for Anthropic provider", async () => {
        // Mock Anthropic /v1/messages endpoint (NOT /chat/completions)
        fetchMock
          .get("https://api.anthropic.com")
          .intercept({
            path: "/v1/messages",
            method: "POST",
          })
          .reply(() => ({
            statusCode: 200,
            data: {
              id: "msg_test",
              type: "message",
              role: "assistant",
              content: [{ type: "text", text: "Test response" }],
              model: "claude-3-7-sonnet-20250219",
              usage: { input_tokens: 10, output_tokens: 5 },
            },
          }))
          .persist();
        
        const response = await SELF.fetch(
          "https://ai-gateway.helicone.ai/v1/chat/completions",
          {
            method: "POST",
            headers: {
              "Content-Type": "application/json",
              Authorization: "Bearer sk-helicone-test",
              "Helicone-Gateway-Body-Mapping": "NO_MAPPING",
            },
            body: JSON.stringify({
              model: "claude-3-7-sonnet-20250219/anthropic",
              messages: [{ role: "user", content: "Test" }],
              max_tokens: 100,
            }),
          }
        );
        
        expect(response.status).toBe(200);
      });
    });
    
    describe("Fallback with comma-separated models", () => {
      it("should try bedrock first, then fallback to anthropic", async () => {
        // Mock Bedrock to fail
        fetchMock
          .get("https://bedrock-runtime.us-east-1.amazonaws.com")
          .intercept({
            path: "/model/us.anthropic.claude-3-7-sonnet-20250219-v1:0/invoke",
            method: "POST",
          })
          .reply(() => ({ statusCode: 500, data: { error: "Bedrock failed" } }))
          .persist();
        
        // Mock Anthropic to succeed (on /v1/messages)
        fetchMock
          .get("https://api.anthropic.com")
          .intercept({
            path: "/v1/messages",
            method: "POST",
          })
          .reply(() => ({
            statusCode: 200,
            data: {
              id: "msg_fallback",
              type: "message",
              role: "assistant",
              content: [{ type: "text", text: "Fallback response" }],
              model: "claude-3-7-sonnet-20250219",
              usage: { input_tokens: 10, output_tokens: 5 },
            },
          }))
          .persist();
        
        const response = await SELF.fetch(
          "https://ai-gateway.helicone.ai/v1/chat/completions",
          {
            method: "POST",
            headers: {
              "Content-Type": "application/json",
              Authorization: "Bearer sk-helicone-test",
            },
            body: JSON.stringify({
              model: "us.anthropic.claude-3-7-sonnet-20250219-v1:0/bedrock,claude-3-7-sonnet-20250219/anthropic",
              messages: [{ role: "user", content: "Test fallback" }],
              max_tokens: 50,
            }),
          }
        );
        
        expect(response.status).toBe(200);
      });
    });
    
    describe("Model with NO_MAPPING and fallback", () => {
      it("should handle Anthropic SDK format with fallback", async () => {
        const model = "claude-3-7-sonnet-20250219/anthropic, us.anthropic.claude-3-7-sonnet-20250219-v1:0/bedrock";
        
        // Mock first model (anthropic) to succeed on /v1/messages
        fetchMock
          .get("https://api.anthropic.com")
          .intercept({
            path: "/v1/messages",
            method: "POST",
          })
          .reply(() => ({
            statusCode: 200,
            data: {
              id: "msg_anthropic_sdk",
              type: "message",
              role: "assistant",
              content: [{ type: "text", text: "Anthropic SDK response" }],
              model: "claude-3-7-sonnet-20250219",
              usage: { input_tokens: 10, output_tokens: 5 },
            },
          }))
          .persist();
        
        const response = await SELF.fetch(
          "https://ai-gateway.helicone.ai/v1/chat/completions",
          {
            method: "POST",
            headers: {
              "Content-Type": "application/json",
              Authorization: "Bearer sk-helicone-test",
              "Helicone-Gateway-Body-Mapping": "NO_MAPPING",
            },
            body: JSON.stringify({
              model: model,
              system: "You are a helpful assistant.",
              messages: [{ role: "user", content: "What is the capital of France?" }],
              max_tokens: 100,
            }),
          }
        );
        
        expect(response.status).toBe(200);
      });
    });
  });
});
