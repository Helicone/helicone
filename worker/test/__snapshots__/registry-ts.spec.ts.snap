// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Registry Tests with TypeScript Configs > Registry Endpoint URLs > should match expected PTB endpoint URLs (snapshot) > registry-endpoint-urls 1`] = `
{
  "chatgpt-4o-latest:openai": {
    "byok": {
      "default": "https://api.openai.com/v1/chat/completions",
      "us-east-1": "https://api.openai.com/v1/chat/completions",
      "us-west-2": "https://api.openai.com/v1/chat/completions",
      "vertex-us-east1": "https://api.openai.com/v1/chat/completions",
    },
    "ptb": {
      "*": "https://api.openai.com/v1/chat/completions",
    },
  },
  "claude-3.5-haiku:anthropic": {
    "byok": {
      "default": "https://api.anthropic.com/v1/chat/completions",
      "us-east-1": "https://api.anthropic.com/v1/chat/completions",
      "us-west-2": "https://api.anthropic.com/v1/chat/completions",
      "vertex-us-east1": "https://api.anthropic.com/v1/chat/completions",
    },
    "ptb": {
      "*": "https://api.anthropic.com/v1/chat/completions",
    },
  },
  "claude-3.5-haiku:bedrock": {
    "byok": {
      "default": "https://bedrock-runtime.us-east-1.amazonaws.com/model/anthropic.claude-3-5-haiku-20241022-v1:0/invoke",
      "us-east-1": "https://bedrock-runtime.us-east-1.amazonaws.com/model/anthropic.claude-3-5-haiku-20241022-v1:0/invoke",
      "us-west-2": "https://bedrock-runtime.us-west-2.amazonaws.com/model/anthropic.claude-3-5-haiku-20241022-v1:0/invoke",
      "vertex-us-east1": "https://bedrock-runtime.us-east1.amazonaws.com/model/anthropic.claude-3-5-haiku-20241022-v1:0/invoke",
    },
    "ptb": {
      "us-east-1": "https://bedrock-runtime.us-east-1.amazonaws.com/model/anthropic.claude-3-5-haiku-20241022-v1:0/invoke",
    },
  },
  "claude-3.5-haiku:vertex": {
    "byok": {
      "vertex-us-east1": "https://us-east1-aiplatform.googleapis.com/v1/projects/test-project/locations/us-east1/publishers/anthropic/models/claude-3-5-haiku@20241022:streamRawPredict",
    },
    "ptb": {},
  },
  "claude-3.5-sonnet-v2:anthropic": {
    "byok": {
      "default": "https://api.anthropic.com/v1/chat/completions",
      "us-east-1": "https://api.anthropic.com/v1/chat/completions",
      "us-west-2": "https://api.anthropic.com/v1/chat/completions",
      "vertex-us-east1": "https://api.anthropic.com/v1/chat/completions",
    },
    "ptb": {
      "*": "https://api.anthropic.com/v1/chat/completions",
    },
  },
  "claude-3.5-sonnet-v2:bedrock": {
    "byok": {
      "default": "https://bedrock-runtime.us-east-1.amazonaws.com/model/anthropic.claude-3-5-sonnet-20241022-v2:0/invoke",
      "us-east-1": "https://bedrock-runtime.us-east-1.amazonaws.com/model/anthropic.claude-3-5-sonnet-20241022-v2:0/invoke",
      "us-west-2": "https://bedrock-runtime.us-west-2.amazonaws.com/model/anthropic.claude-3-5-sonnet-20241022-v2:0/invoke",
      "vertex-us-east1": "https://bedrock-runtime.us-east1.amazonaws.com/model/anthropic.claude-3-5-sonnet-20241022-v2:0/invoke",
    },
    "ptb": {
      "us-east-1": "https://bedrock-runtime.us-east-1.amazonaws.com/model/anthropic.claude-3-5-sonnet-20241022-v2:0/invoke",
    },
  },
  "claude-3.5-sonnet-v2:vertex": {
    "byok": {
      "vertex-us-east1": "https://us-east1-aiplatform.googleapis.com/v1/projects/test-project/locations/us-east1/publishers/anthropic/models/claude-3-5-sonnet-v2@20241022:streamRawPredict",
    },
    "ptb": {
      "global": "",
    },
  },
  "claude-3.7-sonnet:anthropic": {
    "byok": {
      "default": "https://api.anthropic.com/v1/chat/completions",
      "us-east-1": "https://api.anthropic.com/v1/chat/completions",
      "us-west-2": "https://api.anthropic.com/v1/chat/completions",
      "vertex-us-east1": "https://api.anthropic.com/v1/chat/completions",
    },
    "ptb": {
      "*": "https://api.anthropic.com/v1/chat/completions",
    },
  },
  "claude-3.7-sonnet:bedrock": {
    "byok": {
      "default": "https://bedrock-runtime.us-east-1.amazonaws.com/model/anthropic.claude-3-7-sonnet-20250219-v1:0/invoke",
      "us-east-1": "https://bedrock-runtime.us-east-1.amazonaws.com/model/anthropic.claude-3-7-sonnet-20250219-v1:0/invoke",
      "us-west-2": "https://bedrock-runtime.us-west-2.amazonaws.com/model/anthropic.claude-3-7-sonnet-20250219-v1:0/invoke",
      "vertex-us-east1": "https://bedrock-runtime.us-east1.amazonaws.com/model/anthropic.claude-3-7-sonnet-20250219-v1:0/invoke",
    },
    "ptb": {
      "us-east-1": "https://bedrock-runtime.us-east-1.amazonaws.com/model/anthropic.claude-3-7-sonnet-20250219-v1:0/invoke",
    },
  },
  "claude-3.7-sonnet:vertex": {
    "byok": {
      "vertex-us-east1": "https://us-east1-aiplatform.googleapis.com/v1/projects/test-project/locations/us-east1/publishers/anthropic/models/claude-3-7-sonnet@20250219:streamRawPredict",
    },
    "ptb": {
      "global": "",
    },
  },
  "claude-opus-4-1:anthropic": {
    "byok": {
      "default": "https://api.anthropic.com/v1/chat/completions",
      "us-east-1": "https://api.anthropic.com/v1/chat/completions",
      "us-west-2": "https://api.anthropic.com/v1/chat/completions",
      "vertex-us-east1": "https://api.anthropic.com/v1/chat/completions",
    },
    "ptb": {
      "*": "https://api.anthropic.com/v1/chat/completions",
    },
  },
  "claude-opus-4-1:bedrock": {
    "byok": {
      "default": "https://bedrock-runtime.us-east-1.amazonaws.com/model/anthropic.claude-opus-4-1-20250805-v1:0/invoke",
      "us-east-1": "https://bedrock-runtime.us-east-1.amazonaws.com/model/anthropic.claude-opus-4-1-20250805-v1:0/invoke",
      "us-west-2": "https://bedrock-runtime.us-west-2.amazonaws.com/model/anthropic.claude-opus-4-1-20250805-v1:0/invoke",
      "vertex-us-east1": "https://bedrock-runtime.us-east1.amazonaws.com/model/anthropic.claude-opus-4-1-20250805-v1:0/invoke",
    },
    "ptb": {
      "us-east-1": "https://bedrock-runtime.us-east-1.amazonaws.com/model/anthropic.claude-opus-4-1-20250805-v1:0/invoke",
    },
  },
  "claude-opus-4-1:vertex": {
    "byok": {
      "vertex-us-east1": "https://us-east1-aiplatform.googleapis.com/v1/projects/test-project/locations/us-east1/publishers/anthropic/models/claude-opus-4-1@20250805:streamRawPredict",
    },
    "ptb": {
      "global": "",
    },
  },
  "claude-opus-4:anthropic": {
    "byok": {
      "default": "https://api.anthropic.com/v1/chat/completions",
      "us-east-1": "https://api.anthropic.com/v1/chat/completions",
      "us-west-2": "https://api.anthropic.com/v1/chat/completions",
      "vertex-us-east1": "https://api.anthropic.com/v1/chat/completions",
    },
    "ptb": {
      "*": "https://api.anthropic.com/v1/chat/completions",
    },
  },
  "claude-opus-4:bedrock": {
    "byok": {
      "default": "https://bedrock-runtime.us-east-1.amazonaws.com/model/anthropic.claude-opus-4-20250514-v1:0/invoke",
      "us-east-1": "https://bedrock-runtime.us-east-1.amazonaws.com/model/anthropic.claude-opus-4-20250514-v1:0/invoke",
      "us-west-2": "https://bedrock-runtime.us-west-2.amazonaws.com/model/anthropic.claude-opus-4-20250514-v1:0/invoke",
      "vertex-us-east1": "https://bedrock-runtime.us-east1.amazonaws.com/model/anthropic.claude-opus-4-20250514-v1:0/invoke",
    },
    "ptb": {
      "us-east-1": "https://bedrock-runtime.us-east-1.amazonaws.com/model/anthropic.claude-opus-4-20250514-v1:0/invoke",
    },
  },
  "claude-opus-4:vertex": {
    "byok": {
      "vertex-us-east1": "https://us-east1-aiplatform.googleapis.com/v1/projects/test-project/locations/us-east1/publishers/anthropic/models/claude-opus-4@20250514:streamRawPredict",
    },
    "ptb": {
      "global": "",
    },
  },
  "claude-sonnet-4:anthropic": {
    "byok": {
      "default": "https://api.anthropic.com/v1/chat/completions",
      "us-east-1": "https://api.anthropic.com/v1/chat/completions",
      "us-west-2": "https://api.anthropic.com/v1/chat/completions",
      "vertex-us-east1": "https://api.anthropic.com/v1/chat/completions",
    },
    "ptb": {
      "*": "https://api.anthropic.com/v1/chat/completions",
    },
  },
  "claude-sonnet-4:bedrock": {
    "byok": {
      "default": "https://bedrock-runtime.us-east-1.amazonaws.com/model/anthropic.claude-sonnet-4-20250514-v1:0/invoke",
      "us-east-1": "https://bedrock-runtime.us-east-1.amazonaws.com/model/anthropic.claude-sonnet-4-20250514-v1:0/invoke",
      "us-west-2": "https://bedrock-runtime.us-west-2.amazonaws.com/model/anthropic.claude-sonnet-4-20250514-v1:0/invoke",
      "vertex-us-east1": "https://bedrock-runtime.us-east1.amazonaws.com/model/anthropic.claude-sonnet-4-20250514-v1:0/invoke",
    },
    "ptb": {
      "us-east-1": "https://bedrock-runtime.us-east-1.amazonaws.com/model/anthropic.claude-sonnet-4-20250514-v1:0/invoke",
    },
  },
  "claude-sonnet-4:vertex": {
    "byok": {
      "vertex-us-east1": "https://us-east1-aiplatform.googleapis.com/v1/projects/test-project/locations/us-east1/publishers/anthropic/models/claude-sonnet-4@20250514:streamRawPredict",
    },
    "ptb": {
      "global": "",
    },
  },
  "gpt-4.1-mini:openai": {
    "byok": {
      "default": "https://api.openai.com/v1/chat/completions",
      "us-east-1": "https://api.openai.com/v1/chat/completions",
      "us-west-2": "https://api.openai.com/v1/chat/completions",
      "vertex-us-east1": "https://api.openai.com/v1/chat/completions",
    },
    "ptb": {
      "*": "https://api.openai.com/v1/chat/completions",
    },
  },
  "gpt-4.1-nano:openai": {
    "byok": {
      "default": "https://api.openai.com/v1/chat/completions",
      "us-east-1": "https://api.openai.com/v1/chat/completions",
      "us-west-2": "https://api.openai.com/v1/chat/completions",
      "vertex-us-east1": "https://api.openai.com/v1/chat/completions",
    },
    "ptb": {
      "*": "https://api.openai.com/v1/chat/completions",
    },
  },
  "gpt-4.1:openai": {
    "byok": {
      "default": "https://api.openai.com/v1/chat/completions",
      "us-east-1": "https://api.openai.com/v1/chat/completions",
      "us-west-2": "https://api.openai.com/v1/chat/completions",
      "vertex-us-east1": "https://api.openai.com/v1/chat/completions",
    },
    "ptb": {
      "*": "https://api.openai.com/v1/chat/completions",
    },
  },
  "gpt-4:openai": {
    "byok": {
      "default": "https://api.openai.com/v1/chat/completions",
      "us-east-1": "https://api.openai.com/v1/chat/completions",
      "us-west-2": "https://api.openai.com/v1/chat/completions",
      "vertex-us-east1": "https://api.openai.com/v1/chat/completions",
    },
    "ptb": {
      "*": "https://api.openai.com/v1/chat/completions",
    },
  },
  "gpt-4o-mini-search-preview:openai": {
    "byok": {
      "default": "https://api.openai.com/v1/chat/completions",
      "us-east-1": "https://api.openai.com/v1/chat/completions",
      "us-west-2": "https://api.openai.com/v1/chat/completions",
      "vertex-us-east1": "https://api.openai.com/v1/chat/completions",
    },
    "ptb": {
      "*": "https://api.openai.com/v1/chat/completions",
    },
  },
  "gpt-4o-mini:openai": {
    "byok": {
      "default": "https://api.openai.com/v1/chat/completions",
      "us-east-1": "https://api.openai.com/v1/chat/completions",
      "us-west-2": "https://api.openai.com/v1/chat/completions",
      "vertex-us-east1": "https://api.openai.com/v1/chat/completions",
    },
    "ptb": {
      "*": "https://api.openai.com/v1/chat/completions",
    },
  },
  "gpt-4o-search-preview:openai": {
    "byok": {
      "default": "https://api.openai.com/v1/chat/completions",
      "us-east-1": "https://api.openai.com/v1/chat/completions",
      "us-west-2": "https://api.openai.com/v1/chat/completions",
      "vertex-us-east1": "https://api.openai.com/v1/chat/completions",
    },
    "ptb": {
      "*": "https://api.openai.com/v1/chat/completions",
    },
  },
  "gpt-4o:extended": {
    "byok": {},
    "ptb": {
      "openai:*": "https://api.openai.com/v1/chat/completions",
    },
  },
  "gpt-4o:openai": {
    "byok": {
      "default": "https://api.openai.com/v1/chat/completions",
      "us-east-1": "https://api.openai.com/v1/chat/completions",
      "us-west-2": "https://api.openai.com/v1/chat/completions",
      "vertex-us-east1": "https://api.openai.com/v1/chat/completions",
    },
    "ptb": {
      "*": "https://api.openai.com/v1/chat/completions",
    },
  },
  "o1-mini:openai": {
    "byok": {
      "default": "https://api.openai.com/v1/chat/completions",
      "us-east-1": "https://api.openai.com/v1/chat/completions",
      "us-west-2": "https://api.openai.com/v1/chat/completions",
      "vertex-us-east1": "https://api.openai.com/v1/chat/completions",
    },
    "ptb": {
      "*": "https://api.openai.com/v1/chat/completions",
    },
  },
  "o1-pro:openai": {
    "byok": {
      "default": "https://api.openai.com/v1/chat/completions",
      "us-east-1": "https://api.openai.com/v1/chat/completions",
      "us-west-2": "https://api.openai.com/v1/chat/completions",
      "vertex-us-east1": "https://api.openai.com/v1/chat/completions",
    },
    "ptb": {
      "*": "https://api.openai.com/v1/chat/completions",
    },
  },
  "o1:openai": {
    "byok": {
      "default": "https://api.openai.com/v1/chat/completions",
      "us-east-1": "https://api.openai.com/v1/chat/completions",
      "us-west-2": "https://api.openai.com/v1/chat/completions",
      "vertex-us-east1": "https://api.openai.com/v1/chat/completions",
    },
    "ptb": {
      "*": "https://api.openai.com/v1/chat/completions",
    },
  },
  "o3-mini-high:openai": {
    "byok": {
      "default": "https://api.openai.com/v1/chat/completions",
      "us-east-1": "https://api.openai.com/v1/chat/completions",
      "us-west-2": "https://api.openai.com/v1/chat/completions",
      "vertex-us-east1": "https://api.openai.com/v1/chat/completions",
    },
    "ptb": {
      "*": "https://api.openai.com/v1/chat/completions",
    },
  },
  "o3-mini:openai": {
    "byok": {
      "default": "https://api.openai.com/v1/chat/completions",
      "us-east-1": "https://api.openai.com/v1/chat/completions",
      "us-west-2": "https://api.openai.com/v1/chat/completions",
      "vertex-us-east1": "https://api.openai.com/v1/chat/completions",
    },
    "ptb": {
      "*": "https://api.openai.com/v1/chat/completions",
    },
  },
  "o3-pro:openai": {
    "byok": {
      "default": "https://api.openai.com/v1/chat/completions",
      "us-east-1": "https://api.openai.com/v1/chat/completions",
      "us-west-2": "https://api.openai.com/v1/chat/completions",
      "vertex-us-east1": "https://api.openai.com/v1/chat/completions",
    },
    "ptb": {
      "*": "https://api.openai.com/v1/chat/completions",
    },
  },
  "o3:openai": {
    "byok": {
      "default": "https://api.openai.com/v1/chat/completions",
      "us-east-1": "https://api.openai.com/v1/chat/completions",
      "us-west-2": "https://api.openai.com/v1/chat/completions",
      "vertex-us-east1": "https://api.openai.com/v1/chat/completions",
    },
    "ptb": {
      "*": "https://api.openai.com/v1/chat/completions",
    },
  },
  "o4-mini-high:openai": {
    "byok": {
      "default": "https://api.openai.com/v1/chat/completions",
      "us-east-1": "https://api.openai.com/v1/chat/completions",
      "us-west-2": "https://api.openai.com/v1/chat/completions",
      "vertex-us-east1": "https://api.openai.com/v1/chat/completions",
    },
    "ptb": {
      "*": "https://api.openai.com/v1/chat/completions",
    },
  },
  "o4-mini:openai": {
    "byok": {
      "default": "https://api.openai.com/v1/chat/completions",
      "us-east-1": "https://api.openai.com/v1/chat/completions",
      "us-west-2": "https://api.openai.com/v1/chat/completions",
      "vertex-us-east1": "https://api.openai.com/v1/chat/completions",
    },
    "ptb": {
      "*": "https://api.openai.com/v1/chat/completions",
    },
  },
}
`;
